app/Actions/Backup/RestoreBackup.php

```
@@ -70,20 +75,52 @@ private function broadcastFileUpdate(BackupFile $backupFile): void
+    private function validate(BackupFile $backupFile, array $input, BackupType $backupType): void
     {
         $rules = [];
 
         if ($backupType === BackupType::DATABASE) {
             $rules['database'] = [
                 'required',
-                Rule::exists('databases', 'id')->where('server_id', $server->id),
+                Rule::exists('databases', 'id')
+                    ->whereNull('deleted_at')
+                    ->whereIn('server_id', $backupFile->backup->server->project->servers()->pluck('id')->all()),
+                function (string $attribute, mixed $value, Closure $fail) use ($backupFile): void {
```

PHPMD (2.15.0)

[warning] Avoid unused parameters such as '$attribute'. (undefined)

(UnusedFormalParameter)